### ISSUE TITLE
Crash: null object reference on init

### ISSUE THREAD

**Issue by kristfal** [cesardeazevedo/react-native-bottom-sheet-behavior#40]
Hey,

We're seeing the crashes using latest beta (with RN 0.51.1). Seems to be a race condition on init, let me know if you need more info. Using `.beta8`.

This code crashes:

```
class App extends React.Component<{}> {
  render() {
    return (
      <CoordinatorLayout style={{flex: 1}}>
        <BottomSheetBehavior
          peekHeight={70}
          hideable={false}
          state={BottomSheetBehavior.STATE_COLLAPSED}
        >
          <View style={{backgroundColor: '#4389f2'}}>
            <View style={{padding: 26}}>
              <Text>BottomSheetBehavior!</Text>
            </View>
            <View style={{height: 200, backgroundColor: '#fff'}} />
          </View>
        </BottomSheetBehavior>
      </CoordinatorLayout>
    );
  }
}

export default App;
```

This code works:

```
class App extends React.Component<{}> {
  constructor() {
    super();
    this.state = {
      ok: false,
    }
  }

  componentDidMount() {
    setTimeout(() => this.setState({ok: true}), 5000);
  }

  render() {
    return (
      <CoordinatorLayout style={{flex: 1}}>
        {this.state.ok && (
          <BottomSheetBehavior
            peekHeight={70}
            hideable={false}
            state={BottomSheetBehavior.STATE_COLLAPSED}
          >
            <View style={{backgroundColor: '#4389f2'}}>
              <View style={{padding: 26}}>
                <Text>BottomSheetBehavior!</Text>
              </View>
              <View style={{height: 200, backgroundColor: '#fff'}} />
            </View>
          </BottomSheetBehavior>
        )}
      </CoordinatorLayout>
    );
  }
}
```

Stack:

```
Attempt to invoke virtual method 'void android.support.v4.widget.ViewDragHelper.process TouhEnvent(android.view.MotionEvent)' on a null object reference

onTouchEvent
RnBottomSheetBehavior.java:273

resetTouchBehaviors
CoordinatorLayout.java:396

onAttachedToWindow
CoordinatorLayout.java:238

dispatchAttachedToWindow
View.java:16638
```

**Comment by cesardeazevedo** [comment]
Hi, thanks for report, i tried to reproduce your code with RN 0.51.1, but i didn't had any crash, maybe i am doing something wrong?

Also, can your share your gradle dependencies in your [build.gradle](https://github.com/cesardeazevedo/react-native-bottom-sheet-behavior/blob/master/example/android/app/build.gradle) file?

**Comment by kristfal** [comment]
Hey,

not sure whats so special about our project. Beta7 had the same behavior. Here is the build gradle btw:

```
dependencies {
    compile project(':react-native-bottom-sheet-behavior')
    compile project(':react-native-simple-compass')
    compile project(':react-native-iap')
    compile project(':react-native-background-fetch')
    compile project(':react-native-background-geolocation-android')
    compile project(':react-native-code-push')
    compile project(':appcenter-crashes')
    compile project(':appcenter-analytics')
    compile project(':appcenter')

    // Mapbox workaround: https://github.com/mapbox/react-native-mapbox-gl/issues/1139
    compile (project(':@mapbox/react-native-mapbox-gl')) {
        compile ('com.squareup.okhttp3:okhttp:3.6.0') {
            force = true
        }
    }

    compile project(':react-native-google-analytics-bridge')
    compile project(':react-native-bottomsheet')
    compile project(':react-native-image-picker')
    compile project(':react-native-linear-gradient')
    compile project(':react-native-orientation')
    compile project(':react-native-interactable')
    compile project(':react-native-system-setting')
    compile project(':react-native-material-kit')
    compile project(':react-native-fbsdk')
    compile project(':react-native-svg')
    compile project(':react-native-i18n')
    compile project(':react-native-vector-icons')
    compile project(':react-native-intercom')
    compile fileTree(dir: "libs", include: ["*.jar"])

    compile "com.android.support:multidex:1.0.1"
    compile "com.android.support:appcompat-v7:$rootProject.supportLibVersion"
    compile "com.facebook.react:react-native:+"
    compile "io.intercom.android:intercom-sdk-base:3.+"

    androidTestImplementation(project(path: ":detox"))
    androidTestImplementation 'junit:junit:4.12'
    androidTestImplementation 'com.android.support.test:runner:1.0.1'
    androidTestImplementation 'com.android.support.test:rules:1.0.1'

    compile project(':react-native-keep-awake')
 }
```

here is our exts as well:

```
ext {
    compileSdkVersion   = 27
    targetSdkVersion    = 27
    buildToolsVersion   = "27.0.3"
    supportLibVersion   = "27.1.1"
    googlePlayServicesVersion = "15.0.0"
}
```

Let me know if there is anything else we can help with.

**Comment by FlorianBergmann** [comment]
According to [this github issue](https://github.com/Krupen/FabulousFilter/issues/9) there is a change when compiling with 27 that leads to `onLayoutChild` being called before `onTouchEvent`. Due to this there will be a NullPointerException thrown in line 273 of `RNBottomSheetBehavior.java:`

> java.lang.NullPointerException: Attempt to invoke virtual method 'void android.support.v4.widget.ViewDragHelper.processTouchEvent(android.view.MotionEvent)' on a null object reference
>         at com.bottomsheetbehavior.RNBottomSheetBehavior.onTouchEvent(RNBottomSheetBehavior.java:273)

I created a pull request with a fix: #41

**Comment by cesardeazevedo** [comment]
I've successfully reproduced your problem when i compiled with 27, i applied @FlorianBergmann patch and worked.

i will make a new release soon.

**Comment by kristfal** [comment]
Perf! Thanks for quick turnaround. We’ll drop our workaround and use the next release.

**Comment by cesardeazevedo** [comment]
Shipped at `1.0.0`
